uxarray/grid/integrate.py

```diff
@@ -52,40 +57,99 @@ def _get_zonal_faces_weight_at_constLat(
+        # Remove the Int_fill_value from the face_edges
+        face_edges = face_edges[np.all(face_edges != INT_FILL_VALUE, axis=(1, 2))]
         if is_face_GCA_list is not None:
             is_GCA_list = is_face_GCA_list[face_index]
         else:
             is_GCA_list = None
         face_interval_df = _get_zonal_face_interval(
             face_edges,
             latitude_cart,
-            face_latlon_bound[face_index],
+            face_latlon_bound_candidate[face_index],
             is_directed=is_directed,
             is_latlonface=is_latlonface,
             is_GCA_list=is_GCA_list,
         )
-        for _, row in face_interval_df.iterrows():
-            intervals_list.append(
-                {"start": row["start"], "end": row["end"], "face_index": face_index}
-            )
+        # If any end of the interval is NaN
+        if face_interval_df.isnull().values.any():
+            # Skip this face as it is just being touched by the constant latitude
+            continue
+        # Check if the DataFrame is empty (start and end are both 0)
+        if (face_interval_df["start"] == 0).all() and (
+            face_interval_df["end"] == 0
+        ).all():
+            # Skip this face as it is just being touched by the constant latitude
+            continue
+        else:
+            for _, row in face_interval_df.iterrows():
+                intervals_list.append(
+                    {"start": row["start"], "end": row["end"], "face_index": face_index}
+                )
 
     intervals_df = pd.DataFrame(intervals_list)
-    overlap_contributions, total_length = _process_overlapped_intervals(intervals_df)
+    try:
+        overlap_contributions, total_length = _process_overlapped_intervals(
+            intervals_df
+        )
 
-    # Calculate weights for each face
-    weights = {
-        face_index: overlap_contributions.get(face_index, 0.0) / total_length
-        for face_index in range(len(faces_edges_cart))
-    }
+        # Calculate weights for each face
+        weights = {
+            face_index: overlap_contributions.get(face_index, 0.0) / total_length
+            for face_index in range(len(faces_edges_cart_candidate))
+        }
 
-    # Convert weights to DataFrame
-    weights_df = pd.DataFrame(list(weights.items()), columns=["face_index", "weight"])
-    return weights_df
+        # Convert weights to DataFrame
+        weights_df = pd.DataFrame(
+            list(weights.items()), columns=["face_index", "weight"]
+        )
+        return weights_df
+
+    except ValueError:
+        print(f"Face index: {face_index}")
+        print(f"Face edges information: {face_edges}")
+        print(f"Constant z0: {latitude_cart}")
+        print(
+            f"Face latlon bound information: {face_latlon_bound_candidate[face_index]}"
+        )
+        print(f"Face interval information: {face_interval_df}")
+        # Handle the exception or propagate it further if necessary
+        raise
```

Comment:
Leftover debugging prints?

Reply:
This is for what happens if an exception occurs. But if you think an user doesn't need this information, you can remove it